test/spec_1_tests.ts

```diff
@@ -97,73 +97,60 @@ describe("CloudEvents Spec v1.0", () => {
     // is transmitted across the wire, this value will be
     // converted to JSON
     it("should be ok when the type is an object", () => {
-      cloudevent["object-extension"] = { some: "object" };
+      cloudevent = cloudevent.cloneWith({ "object-extension": { some: "object" } });
       expect(cloudevent.validate()).to.equal(true);
     });
   });
 
   describe("The Constraints check", () => {
     describe("'id'", () => {
-      it("should throw an error when is absent", () => {
-        delete cloudevent.id;
-        expect(cloudevent.validate.bind(cloudevent)).to.throw(ValidationError, "invalid payload");
-        cloudevent.id = id;
+      it("should throw an error when trying to remove", () => {
+        expect(() => {
+          delete cloudevent.id;
+        }).to.throw();
       });
 
-      it("should throw an error when is empty", () => {
-        cloudevent.id = "";
-        expect(cloudevent.validate.bind(cloudevent)).to.throw(ValidationError, "invalid payload");
-        cloudevent.id = id;
+      it("defaut ID create when an empty string", () => {
+        cloudevent = cloudevent.cloneWith({ id: "" });
+        expect(cloudevent.id.length).to.be.greaterThan(0);
       });
     });
 
     describe("'source'", () => {
-      it("should throw an error when is absent", () => {
-        delete cloudevent.source;
-        expect(cloudevent.validate.bind(cloudevent)).to.throw(ValidationError, "invalid payload");
-        cloudevent.source = source;
+      it("should throw an error when trying to remove", () => {
+        expect(() => {
+          delete cloudevent.source;
+        }).to.throw();
```

Comment:
```suggestion
        }).to.throw(ValidationError, "invalid payload");
```

Reply:
They actually throw a TypeError instead of a ValidationError,  since the error now comes from deleting a property on a frozen object.

I can still update the tests to check for the TypeError Throw